tools/build/configure.py

```diff
@@ -683,7 +683,7 @@ def build(
                     elif "os" in entry.src_paths[0].parts:  # libultra
                         cflags = ""
                     else:  # papermario
-                        cflags = "-fforce-addr"
+                        cflags = f"-fforce-addr -fmacro-prefix-map={str(entry.src_paths[0]).rsplit('/', 1)[0]}/="
```

Comment:
Forcing `__FILE__` to just be the basename of the file path may break `debug_print`. They are hashed using the `__FILE__` and `__LINE__`. A collision would occur if the basenames are the same, even if they are in different directories, which isn't great.

Perhaps you can make them relative to `src` or the repo root instead? Not sure how much space we have to work with here

Reply:
Perhaps the debug macro should be using FILE_NAME; then we'd be OK if the FILE assert is used only for effects and FILE_NAME is used elsewhere.

Reply:
Hmm, in that case it's probably better to only fully truncate it in effects. We have absolutely no space there, as the largest effect is exactly 0x1000 bytes. By default it's relative to the repo root, which isn't bad in most places, but even that is too much for effects.

Reply:
In that case I suggest appending `cflags += f" -D__FILE_NAME__={entry.obj_path.name} " [around here](https://github.com/bates64/papermario-dx/blob/85f1851c39e9dad78003b8cadc874703b7044461/tools/build/configure.py#L721). This will hopefully have the same behaviour as gcc 12